### PR TITLE
Add base_ring_type methods, remove redundant base_ring methods

### PR DESCRIPTION
**Pull request by fingolfin**
Also fix the `base_ring` for QQBar

### CODE REVIEW

File: src/HeckeMoreStuff.jl [hunk 1]
```diff
@@ -1718,25 +1718,3 @@ end
...
-
-base_ring_type(::Type{fqPolyRepPolyRing}) = fqPolyRepField
-
-base_ring_type(::Type{FqPolyRepPolyRing}) = FqPolyRepField
-
-base_ring_type(::Type{FpPolyRing}) = Nemo.FpField
-
-base_ring_type(::Type{fpPolyRing}) = Nemo.fpField
-
-base_ring_type(::Type{zzModPolyRing}) = Nemo.zzModRing
```
Comment (fingolfin) [inline src/HeckeMoreStuff.jl:1742]: These are redundant, AA handles all `PolyRing` subtypes generically

File: src/antic/nf_elem.jl [hunk 2]
```diff
@@ -21,20 +21,10 @@ parent(a::AbsSimpleNumFieldElem) = a.parent
...
-Returns `Union{}` since a number field doesn't depend on any ring.
-"""
 base_ring(a::AbsSimpleNumField) = Union{}
 
-@doc raw"""
-    base_ring(a::AbsSimpleNumFieldElem)
-
-Returns `Union{}` since a number field doesn't depend on any ring.
-"""
-base_ring(a::AbsSimpleNumFieldElem) = Union{}
```
Comment (fingolfin) [inline src/antic/nf_elem.jl:36]: redundant: already handled by AA

File: src/calcium/qqbar.jl [hunk 3]
```diff
@@ -16,9 +16,9 @@ parent_type(::Type{QQBarFieldElem}) = QQBarField
 
 elem_type(::Type{QQBarField}) = QQBarFieldElem
 
-base_ring(a::QQBarField) = CalciumQQBar
+base_ring_type(::Type{QQBarField}) = typeof(Union{})
 
-base_ring(a::QQBarFieldElem) = CalciumQQBar
+base_ring(::QQBarField) = Union{}
```
Comment (fingolfin) [inline src/calcium/qqbar.jl:21]: Fix: a ring can not be its own base ring

**Review by lgoettgens — APPROVED** [review]
The nightly failure is due to https://github.com/Nemocas/Nemo.jl/issues/1734. Let's ignore this for now

**Pull request merged**